tuna.go

```diff
@@ -74,6 +75,7 @@ type Common struct {
 	SubscriptionPrefix  string
 	Reverse             bool
 	ReverseMetadata     *Metadata
+	EntryServiceInfo    *EntryServiceInfo
```

Comment:
This should not be called EntryServiceInfo because it's used by both entry and exit.
Also since you use it as a replacement for `common.EntryToExitMaxPrice` and `common.ExitToEntryMaxPrice`, you should remove these two fields